src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -141,6 +141,10 @@ public static MemoryMappedFile CreateFromFile(string path, FileMode mode, string
             {
                 throw new ArgumentException(SR.Argument_NewMMFAppendModeNotAllowed, nameof(mode));
             }
+            if (mode == FileMode.Truncate)
+            {
+                throw new ArgumentException(SR.Argument_NewMMFTruncateModeNotAllowed, nameof(mode));
```

Comment:
The ArgumentException from FileMode.Truncate was coming from the [FileStream constructor where it asserts that FileMode.Truncate requires FileSystemRights.Write](https://referencesource.microsoft.com/#mscorlib/system/io/filestream.cs,687), but we were [only passing FileSystemRights.WriteData](https://referencesource.microsoft.com/#System.Core/System/IO/MemoryMappedFiles/MemoryMappedFile.cs,727). This meant that all cases where FileMode == Truncate were causing ArgumentExceptions from inadequate FileSystemRights.

Since we don't have ACL in the netcore FileStream Init function, the only check it has like this [is against the FileAccess](https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/IO/FileStream.cs#L174) which we don't hit because [we actually pass the correct FileAccess for writing](https://github.com/dotnet/corefx/blob/master/src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs#L509). This is also the case in netfx, so the behavior around FileAccess is correct, just inadequate because of the lack of FileSystemRights.

An argument could be made for leaving the behavior as-is and allowing FileMode.Truncate since we don't have FileSystemRights, but it should be done intentionally with a paper trail since it is a behavioral change from netfx.